js/source/worker_tile.js

```diff
@@ -98,10 +104,15 @@ WorkerTile.prototype.parse = function(data, layers, actor, callback) {
         symbolBuckets = this.symbolBuckets = [],
         otherBuckets = [];
 
+    featureTree.numberToLayerIDs = [];
+
     for (var id in bucketsById) {
         bucket = bucketsById[id];
         if (bucket.features.length === 0) continue;
 
+        bucket.index = featureTree.numberToLayerIDs.length;
+        featureTree.numberToLayerIDs.push(bucket.layerIDs);
```

Comment:
`bucket.id` is a string. We could index buckets by a number. It gets trickier for async queries where we need to transfer everything back to workers. It helps if featureTree is self-contained and doesn't need buckets. I made it slightly cleaner here: fa0689f02b1acb5a338142026e1b05b2b0392a93